### PR TITLE
versions: move cmake and musl version into versions.yaml

### PR DESCRIPTION
**Pull request by yyyeerbo**
Fixes: #2162

Signed-off-by: Yang Bo <bo@hyper.sh>

### CODE REVIEW

File: versions.yaml [hunk 1]
```diff
@@ -263,6 +263,32 @@ externals:
...
+    description: |
+      musl is used to build rust agent.
+    url: "https://www.musl-libc.org/"
+    uscan-url: >-
+      https://www.musl-libc.org/releases/
+      musl-([\d\.]+)\.tar\.gz
+    version: "1.1.23"
+    meta:
+      description: |
+        'newest-version' is the latest version works.
```
Comment (jodh-intel) [inline versions.yaml]: Nit: `s/version works/version known to work/` maybe.

Comment (yyyeerbo) [reply versions.yaml]: updated -_-

**Comment by jodh-intel** [comment]
Restarted Fedora CI which barfed on:

```
not ok 1 Check CPU constraints
09:22:33 # (in test file k8s-cpu-ns.bats, line 35)
09:22:33 #   `[ $total_cpus_container -eq $total_cpus ]' failed
09:22:33 # INFO: k8s configured to use runtimeclass
09:22:33 # pod/constraints-cpu-test created
09:22:33 # pod/constraints-cpu-test condition met
09:22:33 # pod "constraints-cpu-test" deleted
09:22:33 Failed at 78: bats "${K8S_TEST_ENTRY}"
```

/cc @GabyCT.

**Pull request merged**